libs/src/sdk/services/Storage/StorageNodeSelector.ts

```diff
@@ -0,0 +1,97 @@
+import { Maybe, RendezvousHash, isNodeHealthy } from '../../../utils'
+import fetch from 'cross-fetch'
+import type { DiscoveryNodeSelector } from '../DiscoveryNodeSelector'
+import type { HealthCheckResponseData } from '../DiscoveryNodeSelector/healthCheckTypes'
+import type { Auth } from '../Auth'
+
+type StorageNode = {
+  endpoint: string
+  ownerDelegateWallet: string
+}
+
+export type StorageNodeSelectorConfig = {
+  bootstrapNodes?: StorageNode[]
+  auth: Auth
+  discoveryNodeSelector?: DiscoveryNodeSelector
+}
+
+export class StorageNodeSelector {
+  private readonly config: StorageNodeSelectorConfig
+  private readonly auth: Auth
+  private nodes: StorageNode[]
+  private orderedNodes?: StorageNode[]
+  private selectedNode?: string | null
+  private readonly discoveryNodeSelector?: DiscoveryNodeSelector
+
+  constructor(config: StorageNodeSelectorConfig) {
+    this.config = config
+    this.auth = this.config.auth
+    this.nodes = this.config.bootstrapNodes ?? []
+    this.discoveryNodeSelector = this.config.discoveryNodeSelector
+
+    this.discoveryNodeSelector?.addEventListener(
+      'change',
+      this.onChangeDiscoveryNode
+    )
+  }
+
+  onChangeDiscoveryNode = async (endpoint: string) => {
+    const healthCheckEndpoint = `${endpoint}/health_check`
+    const discoveryHealthCheckResponse = await fetch(healthCheckEndpoint)
+    if (!discoveryHealthCheckResponse.ok) return
+
+    const responseData: { data: HealthCheckResponseData } =
+      await discoveryHealthCheckResponse.json()
+    const contentNodes = responseData.data.network?.content_nodes
```

Comment:
so how i thought it was supposed to go is 1. get bootstrap nodes from config, 2. get nodes from discovery health and replace them. and if selectedNode is already selected it doesn't switch, it just changes underlying nodes that it will then go to if a reselect is called. does that seem right?

Reply:
Ah gotcha, so we're bootstrapping Storage nodes from config too? If so then yeah this is great